python/pyspark/pandas/groupby.py

```diff
@@ -322,8 +323,12 @@ def aggregate(
                 psdf = psdf.reset_index(level=should_drop_index, drop=drop)
             if len(should_drop_index) < len(self._groupkeys):
                 psdf = psdf.reset_index()
+            index_cols = [c for c in list(psdf.columns) if c not in index_cols]
```

Comment:
ditto. `index_cols = [c for c in psdf._internal.data_spark_column_names if c not in index_cols]`

Reply:
Updated this is well with psdf._internal.column_labels.